packages/e2e-tests/specs/editor/various/copy-cut-paste-whole-blocks.test.js

```diff
@@ -92,4 +92,91 @@ describe( 'Copy/cut/paste of whole blocks', () => {
 		await pressKeyWithModifier( 'primary', 'v' );
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'should handle paste events once', async () => {
+		// Add group block with paragraph
+		await insertBlock( 'Group' );
+		await page.click( '.block-editor-button-block-appender' );
+		await page.click( '.editor-block-list-item-paragraph' );
+		await page.keyboard.type( 'P' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+		// Cut group
+		await pressKeyWithModifier( 'primary', 'x' );
+		await page.keyboard.press( 'Enter' );
+
+		await page.evaluate( () => {
+			window.e2eTestPasteOnce = [];
+			let oldBlocks = wp.data.select( 'core/block-editor' ).getBlocks();
+			wp.data.subscribe( () => {
+				const blocks = wp.data
+					.select( 'core/block-editor' )
+					.getBlocks();
+				if ( blocks !== oldBlocks ) {
+					window.e2eTestPasteOnce.push(
+						blocks.map( ( { clientId, name } ) => ( {
+							clientId,
+							name,
+						} ) )
+					);
+				}
+				oldBlocks = blocks;
+			} );
+		} );
```

Comment:
I wonder how useful testing for this is, if there's no consequences for the content, it doesn't really matter how many times blocks are replaced. It's mostly a purity thing that doesn't really justify all the additional test code to maintain.

Reply:
Yes, this is a bit fragile, but I did want some tests to document the behavior since it's a bit of an odd side effect to consider. I'll see if I can get the E2Es to be stable and we can revisit if its flaky. 

Overall, fixing this is more minor though I do see that this branch also fixes https://github.com/WordPress/gutenberg/issues/34177